mesonbuild/interpreter/interpreter.py

```diff
@@ -3482,6 +3488,11 @@ def build_target(self, node: mparser.BaseNode, args: T.Tuple[str, SourcesVarargs
         target = targetclass(name, self.subdir, self.subproject, for_machine, srcs, struct, objs,
                              self.environment, self.compilers[for_machine], kwargs)
 
+        if len(kwargs['rename']) not in {0, 1, len(target.outputs)}:
```

Comment:
Does this still have the soname issue that was mentioned in earlier attempts?

Reply:
Probably. I don't need that case to work for my projects, and we have had zero traction in getting build subdirectory support merged in meson, so I'm trying a simpler patch to see if I can at least get my use case working without thousands (literally) of warnings. To fix the soname issue, it might be easier to fix up the existing `-soname` code paths to permit a custom name for that too?